### PR TITLE
Creating JavaDocs - Closes Issue #129

### PR DESCRIPTION
**Pull request by SwindleA**
Created JavaDocs for relevant files with in micronautpi4j-utils/src/main/java/com/opensourcewithslu. 

Some observations:

1. There are a lot of repetitive methods for the configuration files in utilities. I think there should be some way to cut down on the repetitiveness of the code. 
2. There are many aspects of the utility files that I had to guess at how they worked. I tried to leave it as open ended so as to not provide false information, just incomplete. I plan on trying to add more details in the remaining week in the sprint.

### CODE REVIEW

**Comment by SwindleA** [comment]
There are only two warnings. They exist because the Micronaut annotation prevents from the javadoc comments from generating. The warnings are in Pi4JFactory, Pi4JMultipinFactory, and Application.java

**Review by ruthvikm — COMMENTED** [review]
Looks good. I didn't see any warnings. Do you have any plan about how to integrate these javadocs in documentation as API References?

**Review by joefol — COMMENTED** [review]
It builds successfully for me and all of the javadocs are visible. I also do not get any error messages relating to the javadocs.

**Comment by SwindleA** [comment]
> Looks good. I didn't see any warnings. Do you have any plan about how to integrate these javadocs in documentation as API References?

The warnings appear when building the javadocs, not the project. 

I do not have a specific plan for how the javadocs will be integrated into the documentation. My only suggestion would be to  link the html file of the javadocs in the adoc files for each of the helper classes.

**Comment by ruthvikm** [comment]
We don't need javadocs folder in our main branch, but we need one in gh-pages. So that the javadocs can be integrated with documentation. This should be automated by writing  a task in the asciidoc-build.yml workflow. 
To integrate that in documentation just create a section called "API References"  for now in index.adoc file with a link that points to javadoc index.html file.

**Review by ruthvikm — APPROVED** [review]
Looks Good

File: micronautpi4j-utils/src/docs/asciidoc/index.adoc [hunk 1]
```diff
@@ -15,6 +15,9 @@
 
 toc::[]
 
+[.text-right]
+https://github.com/oss-slu/Pi4Micronaut/edit/develop/micronautpi4j-utils/src/docs/javadoc/index.html[API References]
```
Comment (ruthvikm) [inline micronautpi4j-utils/src/docs/asciidoc/index.adoc:19]: Why is it pointing to develop branch?

Comment (SwindleA) [reply micronautpi4j-utils/src/docs/asciidoc/index.adoc]: that was a mistake. I fixed it.

**Pull request merged**